components/build/src/lib.rs

```diff
@@ -163,23 +192,19 @@ where
     let proto_files = get_proto_files(&input_dir);
     generate_mod_rs(&out_dir, &proto_files, &mod_file_name.as_ref());
 
-    let includes = includes.into_iter().collect::<Vec<_>>();
+    let includes = includes.iter().collect::<Vec<_>>();
     // Converts paths to strings and adds input dir to includes.
-    let mut includes = includes
-        .iter()
-        .map(|s| {
-            s.as_ref()
-                .to_str()
-                .expect("Include dir name is not convertible to &str")
-        })
-        .collect::<Vec<_>>();
+    let mut includes = includes.iter().map(|s| s.path()).collect::<Vec<_>>();
     includes.push(
         input_dir
             .as_ref()
             .to_str()
-            .expect("Input dir name is not convertible to &str"),
+            .expect("Input dir name is not convertible to &str")
+            .into(),
     );
 
+    let includes: Vec<&str> = includes.iter().map(|s| &**s).collect();
```

Comment:
Do we have to `collect` three times in a row?

Reply:
No, removed 3rd collect, trying to remove 2nd.

Reply:
```suggestion
    let includes: Vec<&str> = includes.iter().map(String::as_str).collect();
```